homeassistant/components/tfl/config_flow.py

```diff
@@ -0,0 +1,284 @@
+"""Config flow for Transport for London integration."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+import logging
+from typing import Any
+from urllib.error import HTTPError, URLError
+
+from tflwrapper import stopPoint
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlow,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import TextSelector, TextSelectorConfig
+
+from .const import CONF_API_APP_KEY, CONF_STOP_POINTS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_API_APP_KEY, default=""): cv.string,
+    }
+)
+STEP_STOP_POINT_DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_STOP_POINTS): TextSelector(TextSelectorConfig(multiple=True))}
+)
+
+
+async def validate_app_key(hass: HomeAssistant, app_key: str) -> None:
+    """Validate the user input for app_key."""
+
+    _LOGGER.debug("Validating app_key")
+    stop_point_api = stopPoint(app_key)
+    # Make a random, cheap, call to the API to validate the app_key
+    try:
+        categories = await hass.async_add_executor_job(stop_point_api.getCategories)
+        _LOGGER.debug("Validating app_key, got categories=%s", categories)
+    except HTTPError as exception:
+        # TfL's API returns a 429 if you pass an invalid app_key, but we also check
+        # for other reasonable error codes in case their behaviour changes
+        error_code = exception.code
+        if error_code in (429, 401, 403):
```

Comment:
The same error codes are also in `__init__.py`, store the tuple in `const.py` instead and use it in both places

Reply:
Tidied up as part of the wider cleanup you called out with the repeated error handling. Only there once now, so I didn't move out to `const.py`